src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -64,6 +64,13 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
     size0 = n
   }
 
+  /** Shrinks the ArrayList to one that is adjusted to the current number of elements in it,
+    * which may replace the array by a shorter one. This allows releasing some unused memory. */
+  def shrinkToSize(): Unit = {
+    if (array.length >= 8 && array.length >= size0 / 2)
```

Comment:
I find the semantics of `reduceSize` very confusing. I thought this logic here looks wrong (shrink if it's more than *half* the required size?) but the size passed to `reduceSize` is actually *twice* the target size.